integration_tests/mockApis/curiousApi.ts

```diff
@@ -74,7 +74,7 @@ export default {
       },
     })
   },
-  stubGetLearnerGoals: (params: LearnerGoalsTestParams) => {
+  stubGetCuriousGoals: (params: LearnerGoalsTestParams) => {
```

Comment:
Where possible I've renamed the terminology "Learner Goals" to "Curious Goals" (because thats a better reflection of what they are / where they come from), and of course we now have "Personal Learning Plan Goals"
I've not done it everywhere as that would be quite a large refactoring, but I've done it where it makes sense in respect of what I'm committing in this PR